proposals/pr-contrib-guide.md

```diff
@@ -0,0 +1,460 @@
+1. Feature requests must have at least 10 :+1: reactions.
+1. The maintainers agree with the implementation plan, including any design
+   discussions (e.g., public interface design, edge cases, backwards
+   compatibility concerns).
+
+If the criteria are not satisfied, the maintainers should explicitly comment
+what criteria are not satsified and how the community can satisfy them. All
+discussions must be resolved prior to making a contribution so that feedback
+during the review process focuses on the implementation of the agreed upon
+design.
+
+Issues that do not receive enough :+1: reactions and do not have any activty for
+a year are considered stale. Without further interaction, they will be closed.
+
+### Contribution-ready
+
+A GitHub issue enters the contribution ready stage when a maintainer publicly
+labels the issue ready for contribution. This stage indicates any user is
+welcome to initiate a contribution via a GitHub pull request. All contribution
+ready issues are listed publicly for the community to discover.
+
+### Implementation
+
+A user can select an issue from the list of contribution ready issues and
+implement their change. When the contributor has completed their implementation,
+they should open a pull request. The maintainers will respond to acknowledge the
+pull request and let the contributor know to expect a preliminary review by the
+within 5 business days to confirm:
+
+1. It is implemented in the manner described in the issue.
+1. It meets the criteria specified in the contribution guide.
+
+If the criteria are not satisfied, the maintainers will comment on the issue
+specifically indicating what work is still required. If the contributor has
+questions, they should refer to the contributing guide documentation or ask the
+maintainers to clarify. Once the contributor makes the required changes, the
+maintainers will follow up within 5 business days.
+
+If the contributor is no longer actively engaged by responding to comments from
+maintainers or making code changes for more than 30 business days, the
+contribution is stale. If not, the pull request will be closed and the
+corresponding issue will be marked as ready for contribution.
+
+### Ready for review
+
+Once the preliminary review criteria from the implementation stage have been met, the
+pull request is added to a queue for maintainer review. This means that the
+maintainers will include the pull request in a prioritization process for
+review. The maintainers will select issues for review based on the following
+criteria:
+
+1. Existing pull request reviews will be completed before starting new reviews.
+1. Older contributions will be selected before more recent ones.
+1. Bugs and documentation changes will be prioritized over feature requests. 
+1. The time available to the maintainers will be considered against the
+   complexity of the contribution.
+1. The maintainers will limit the number of active reviews to [TBD].
+
```

Comment:
Another bullet point to consider, maybe it goes with the cost-benefit factor, but I think we'd also prioritize based on who the contributor is. Like if they are regular contributors that regularly require small feedback cycles, I'd prefer those PRs over someone who is brand new and may take quite a bit of cycles.

I'm not sure if I'm totally convinced we should add it but maybe it is one of the lower important ones.

Reply:
After thinking about it more, I disagree. There will be attrition in new users who graduate to being more frequent/impactful contributors, but ultimately it will increase the strength of the community to encourage return contributors than to take requests from people we already know.

Seasoned contributors should be recognized, and have access to more responsibility. One possibility would be considering their pre-review or review with a higher weight to speed up the process.